src/soep_preparation/clean_variables/pgen.py

```diff
@@ -0,0 +1,220 @@
+from soep_preparation.utilities.data_manipulator import (
+    apply_smallest_int_dtype,
+    create_dummy,
+    object_to_bool_categorical,
+    object_to_float,
+    object_to_int_categorical,
+    object_to_str_categorical,
+)
+
+
+def _education(
+    casmin: "pd.Series[pd.Categorical]",
+    isced: "pd.Series[pd.Categorical]",
+) -> "pd.Series[pd.Categorical]":
+    """Transform education variable to three levels."""
+    out = casmin.combine_first(isced).map(
+        {
+            "in school": "PRIMARY_AND_LOWER_SECONDARY",
+            "inadequately completed": "PRIMARY_AND_LOWER_SECONDARY",
+            "general elementary school": "PRIMARY_AND_LOWER_SECONDARY",
+            "basic vocational qualification": "PRIMARY_AND_LOWER_SECONDARY",
+            "intermediate vocational": "UPPER_SECONDARY",
+            "intermediate general qualification": "UPPER_SECONDARY",
+            "general maturity certificate": "UPPER_SECONDARY",
+            "vocational maturity certificate": "UPPER_SECONDARY",
+            "lower tertiary education": "TERTIARY",
+            "higher tertiary education": "TERTIARY",
+            "Primary education": "PRIMARY_AND_LOWER_SECONDARY",
+            "Lower secondary education": "PRIMARY_AND_LOWER_SECONDARY",
+            "Upper secondary education": "UPPER_SECONDARY",
+            "Post-secondary non-tertiary education": "UPPER_SECONDARY",
+            "Short-cycle tertiary education": "UPPER_SECONDARY",
+            "Bachelor s or equivalent level": "TERTIARY",
+            "Master s or equivalent level": "TERTIARY",
+            "Doctoral or equivalent level": "TERTIARY",
+        },
+    )
+    cat_type = pd.CategoricalDtype(
+        categories=pd.Series(
+            ["PRIMARY_AND_LOWER_SECONDARY", "UPPER_SECONDARY", "TERTIARY"],
+        ).astype("str[pyarrow]"),
+        ordered=True,
+    )
+    return out.astype(cat_type)
+
+
+def _in_education(
+    employment: "pd.Series[pd.Categorical]",
+    occupation: "pd.Series[pd.Categorical]",
+) -> "pd.Series[pd.Categorical]":
+    in_education = [
+        "Aspiranten (1990 Ost)",
+        "Auszubildende (1984-1999), Lehrlinge (1990 Ost)",
+        "Auszubildende, gewerblich-technisch (ab 2000)",
+        "Auszubildende, kaufmännisch (ab 2000)",
+        "NE: in Ausbildung, inkl. Weiterbildung, Berufsausbildung, Lehre",
+        "Volontäre, Praktikanten",
+    ]
+    out = create_dummy(employment, "Ausbildung, Lehre")
+    # Set in_education to missing if out of the labor force -- could mean anything.
+    out = out.where(employment != "Nicht erwerbstätig", pd.NA)
+    return out.fillna(create_dummy(occupation, in_education, "isin"))
+
+
+def _self_employed_occupations(
+    occupation: "pd.Series[pd.Categorical]",
+) -> list:
+    """Occupation names that indicate self employment."""
+    occupation_names = list(occupation.dropna().unique())
+    occupations_of_interest = re.compile("^.*Freiberufler.*$|^.*selbstä.*$")
```

Comment:
```suggestion
    occupations_of_interest = re.compile("^.*Freiberufler.*$|^.*selb(st)?stä.*$")
```
Please make the regex robust to spelling according to rules present at various points during the sample period... My guess is untested!

Reply:
Only the spelling "selbständig" is present in the data. I adapted the regex to accompany potential future changes.